safe/gis/raster/contour.py

```diff
@@ -0,0 +1,527 @@
+            '{output_file_path}'.format(output_file_path=output_file_path)))
+
+    return output_file_path
+
+
+def shakemap_contour(shakemap_layer_path, output_file_path='', active_band=1):
+    """Creating contour from a shakemap layer.
+
+    :param shakemap_layer_path: The shake map raster layer path.
+    :type shakemap_layer_path: basestring
+
+    :param output_file_path: The path where the contour will be saved.
+    :type output_file_path: basestring
+
+    :param active_band: The band which the data located, default to 1.
+    :type active_band: int
+
+    :returns: The contour of the shake map layer path.
+    :rtype: basestring
+    """
+    # Set output path
+    if not output_file_path:
+        output_file_path = unique_filename(suffix='.shp', dir=temp_dir())
+
+    output_directory = os.path.dirname(output_file_path)
+    output_file_name = os.path.basename(output_file_path)
+    output_base_name = os.path.splitext(output_file_name)[0]
+
+    # Based largely on
+    # http://svn.osgeo.org/gdal/trunk/autotest/alg/contour.py
+    driver = ogr.GetDriverByName('ESRI Shapefile')
+    ogr_dataset = driver.CreateDataSource(output_file_path)
+    if ogr_dataset is None:
+        # Probably the file existed and could not be overriden
+        raise ContourCreationError(
+            'Could not create datasource for:\n%s. Check that the file '
+            'does not already exist and that you do not have file system '
+            'permissions issues' % output_file_path)
+    layer = ogr_dataset.CreateLayer('contour')
+    field_definition = ogr.FieldDefn('ID', ogr.OFTInteger)
+    layer.CreateField(field_definition)
+    field_definition = ogr.FieldDefn('MMI', ogr.OFTReal)
+    layer.CreateField(field_definition)
+    # So we can fix the x pos to the same x coord as centroid of the
+    # feature so labels line up nicely vertically
+    field_definition = ogr.FieldDefn('X', ogr.OFTReal)
+    layer.CreateField(field_definition)
+    # So we can fix the y pos to the min y coord of the whole contour so
+    # labels line up nicely vertically
+    field_definition = ogr.FieldDefn('Y', ogr.OFTReal)
+    layer.CreateField(field_definition)
+    # So that we can set the html hex colour based on its MMI class
+    field_definition = ogr.FieldDefn('RGB', ogr.OFTString)
+    layer.CreateField(field_definition)
+    # So that we can set the label in it roman numeral form
+    field_definition = ogr.FieldDefn('ROMAN', ogr.OFTString)
+    layer.CreateField(field_definition)
+    # So that we can set the label horizontal alignment
+    field_definition = ogr.FieldDefn('ALIGN', ogr.OFTString)
+    layer.CreateField(field_definition)
+    # So that we can set the label vertical alignment
+    field_definition = ogr.FieldDefn('VALIGN', ogr.OFTString)
+    layer.CreateField(field_definition)
+    # So that we can set feature length to filter out small features
+    field_definition = ogr.FieldDefn('LEN', ogr.OFTReal)
```

Comment:
Since 4.0, all our fields are defined from definitions. Why not the same behavior for X, Y, RGB, Roman, Align, etc ?
You added new definitions about these fields but you are not using them, so we might change these definitions later, it won't make any effects in this code.

Reply:
Yes, I want to do that also. I already tried it actually. But this one is using `ogr`, I can't use the `create_field_from_definition` method. Perhaps I will create new one for `ogr`. But yes, I agree with field creation from the definitions.

Reply:
Yes I saw, we can add one for OGR, it would be the best option. But at least we should read the dictionary for the name

Reply:
BTW, maybe in QGIS 3, we will not use OGR to create the dataset, so maybe just read the dictionary. We will be able to read the numpy array from the QGIS api.

Reply:
Ouch, I just created the function.